src/main/java/com/twilio/http/NetworkHttpClient.java

```diff
@@ -4,47 +4,73 @@
-import org.apache.http.HttpEntity;
-import org.apache.http.HttpHeaders;
-import org.apache.http.HttpResponse;
-import org.apache.http.HttpVersion;
-import org.apache.http.client.utils.HttpClientUtils;
-import org.apache.http.entity.BufferedHttpEntity;
+import org.apache.http.*;
 import org.apache.http.client.config.RequestConfig;
 import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.client.utils.HttpClientUtils;
+import org.apache.http.conn.ConnectionKeepAliveStrategy;
+import org.apache.http.entity.BufferedHttpEntity;
 import org.apache.http.impl.client.HttpClientBuilder;
 import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
 import org.apache.http.message.BasicHeader;
+import org.apache.http.message.BasicHeaderElementIterator;
+import org.apache.http.protocol.HTTP;
+import org.apache.http.protocol.HttpContext;
 
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
-import java.util.Collection;
 import java.util.List;
 import java.util.Map;
 
 public class NetworkHttpClient extends HttpClient {
 
-    private static final int CONNECTION_TIMEOUT = 10000;
-    private static final int SOCKET_TIMEOUT = 30500;
+    public static final int CONNECTION_TIMEOUT = 10000;
+
+    public static final int SOCKET_TIMEOUT = 30500;
+
+    public static final int DEFAULT_MAX_PER_ROUTE = 10;
+
+    public static final int MAX_TOTAL = 20;
+
+    public static final int VALIDATE_AFTER_INACTIVITY = 1000;
+
+    public static final int KEEP_ALIVE_TIMEOUT = 30 * 1000;
 
     private final org.apache.http.client.HttpClient client;
 
     /**
      * Create a new HTTP Client.
      */
     public NetworkHttpClient() {
-        RequestConfig config = RequestConfig.custom()
-            .setConnectTimeout(CONNECTION_TIMEOUT)
-            .setSocketTimeout(SOCKET_TIMEOUT)
-            .build();
+        this(defaultHttpClientBuilder());
+    }
+
+    /**
+     * Create a new HTTP Client using custom configuration.
+     * @param clientBuilder an HttpClientBuilder.
+     */
+    public NetworkHttpClient(HttpClientBuilder clientBuilder) {
+        client = clientBuilder
+                .setDefaultHeaders(twilioHeaders())
+                .build();
+    }
 
-        Collection<Header> headers = Lists.<Header>newArrayList(
-            new BasicHeader("X-Twilio-Client", "java-" + Twilio.VERSION),
-            new BasicHeader(HttpHeaders.USER_AGENT, "twilio-java/" + Twilio.VERSION + " (" + Twilio.JAVA_VERSION + ")"),
-            new BasicHeader(HttpHeaders.ACCEPT, "application/json"),
-            new BasicHeader(HttpHeaders.ACCEPT_ENCODING, "utf-8")
-        );
+    private static List<Header> twilioHeaders() {
+        return Lists.<Header>newArrayList(
+                    new BasicHeader("X-Twilio-Client", "java-" + Twilio.VERSION),
+                    new BasicHeader(
+                            HttpHeaders.USER_AGENT, "twilio-java/" + Twilio.VERSION + " (" + Twilio.JAVA_VERSION + ") custom"
```

Comment:
I say drop `custom` or refactor since this is used for both custom and the default client builder.